ipn/ipnlocal/local.go

```diff
@@ -6395,12 +6433,13 @@ var ErrUnableToSuggestLastExitNode = errors.New("unable to suggest last exit nod
 // Currently, peers with a DERP home are preferred over those without (typically this means Mullvad).
 // Peers are selected based on having a DERP home that is the lowest latency to this device. For peers
 // without a DERP home, we look for geographic proximity to this device's DERP home.
-func (b *LocalBackend) SuggestExitNode() (response apitype.ExitNodeSuggestionResponse, err error) {
-	b.mu.Lock()
+// b.mu must be held.
+func (b *LocalBackend) suggestExitNodeLocked(isAuto bool) (response apitype.ExitNodeSuggestionResponse, err error) {
```

Comment:
@bradfitz I started using the suggestExitNode function in places where the LocalBackend mutex is already held. Is this refactor okay for how I want to use the mutex w/ respect to reading and writing the lastSuggestedExitNode value?